src/components/Switch/index.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { BaseProps } from '../types';
+import { SwitchWrapper, StyledSwitch, SwitchText } from './styled';
+import { COLOR_BRAND, COLOR_GRAY_2 } from '../../styles/colors';
+
+interface Props extends BaseProps {
+    label?: string;
+    onChange?: () => void;
+    disabled?: boolean;
+    value?: boolean;
+}
+
+/**
+ * Switch toggle is a checkable input that communicates if an option is true or false
+ */
+
+const SwitchComponent: React.FC<Props> = ({
+    label,
+    onChange,
+    disabled,
+    value,
+}) => {
+    return (
+        <SwitchWrapper>
+            <StyledSwitch
+                onValueChange={onChange}
+                value={value}
+                trackColor={{ false: COLOR_GRAY_2, true: COLOR_BRAND }}
```

Comment:
In order to keep the separation of concerns let's put this in a const to avoid mix js with jsx